lib/logstash/pipeline.rb

```diff
@@ -252,6 +252,7 @@ def outputworker
   #
   # This method is intended to be called from another thread
   def shutdown
+    Thread.new { report_inflight_events(@input_to_filter, @filter_to_output, @outputs) }
```

Comment:
I see, I like to be very upfront/explicit about creating threads, but I'll think of a better way to design this.